app/uk/gov/hmrc/helptosavetestadminfrontend/repos/VerifiedEmailMongoRepository.scala

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosavetestadminfrontend.repos
+
+import cats.Applicative
+import cats.syntax.traverse._
+import cats.instances.list._
+import cats.syntax.either._
+import com.google.inject.{ImplementedBy, Inject, Singleton}
+import play.api.libs.json.{JsString, Json}
+import play.modules.reactivemongo.ReactiveMongoComponent
+import reactivemongo.api.commands.MultiBulkWriteResult
+import reactivemongo.bson.{BSON, BSONDocument, BSONObjectID, BSONString}
+import uk.gov.hmrc.mongo.ReactiveRepository
+import uk.gov.hmrc.mongo.json.ReactiveMongoFormats
+
+import scala.concurrent.{ExecutionContext, Future}
+
+case class Email(email: String)
+
+object Email {
+  implicit val format = Json.format[Email]
+}
+
+@ImplementedBy(classOf[VerifiedEmailMongoRepositoryImpl])
+trait VerifiedEmailMongoRepository {
+
+  def deleteEmails(emails: List[String])(implicit ec: ExecutionContext): Future[Either[List[String], Unit]]
+
+}
+
+@Singleton
+class VerifiedEmailMongoRepositoryImpl @Inject()(mongo:   ReactiveMongoComponent)
+  extends ReactiveRepository[Email, BSONObjectID](
+    collectionName = "verifiedEmail",
+    mongo = mongo.mongoConnector.db,
+    domainFormat = Json.format[Email],
+    idFormat = ReactiveMongoFormats.objectIdFormats
+  ) with VerifiedEmailMongoRepository {
+
+
+  def deleteEmails(emails: List[String])(implicit ec: ExecutionContext): Future[Either[List[String], Unit]] = {
+
+    val result: List[Future[Either[String, Unit]]] =  emails.map { email ⇒
+      remove("email" -> email).map { res ⇒
+        if (res.hasErrors) {
+          Left(s"An error has occurred while deleting email: $email")
```

Comment:
you are not returning what the actual error is ?

Reply:
good point, I have updated the string to include the errmsg